xhttp/client.go

```diff
@@ -23,6 +23,14 @@ type (
 	}
 	// RetrierOption is used to configure retrier clients created with [NewRetrierClient].
 	RetrierOption func(*retrierClient)
+
+	// RetrierOnRequestDoneFunc is the callback called when using [RetrierWithOnRequestDone].
+	// The [*http.Request] is the original http request that just finished.
+	// The [*http.Response] is the response of the request or nil if the request failed with an error (then the error will be non-nil).
```

Comment:
```suggestion
	// The [*http.Response] is the response of the request or nil if the request failed with an error.
```
I think this goes without saying.

Reply:
I will improve those docs and how invariants are documented, seems like just redirecting what is returned by Do makes more sense than trying to re-specify invariants.